pkg/seedadmission/extension_crds.go

```diff
@@ -0,0 +1,155 @@
+	case
+		metav1.GroupVersionResource{Group: apiextensionsv1beta1.SchemeGroupVersion.Group, Version: apiextensionsv1beta1.SchemeGroupVersion.Version, Resource: "customresourcedefinitions"},
+		metav1.GroupVersionResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Version: apiextensionsv1.SchemeGroupVersion.Version, Resource: "customresourcedefinitions"},
+
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "backupbuckets"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "backupentries"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "controlplanes"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "extensions"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "infrastructures"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "networks"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "operatingsystemconfigs"},
+		metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "workers"}:
+	default:
+		return nil
+	}
+
+	// Older Kubernetes versions don't provide the object neither in OldObject nor in the Object field. In this case
+	// we have to look it up ourselves.
+	var (
+		obj       runtime.Object
+		err       error
+		operation = "DELETE"
+	)
+
+	switch {
+	case request.OldObject.Raw != nil:
+		o := &unstructured.Unstructured{}
+		err = json.Unmarshal(request.OldObject.Raw, o)
+		obj = o
+
+	case request.Object.Raw != nil:
+		o := &unstructured.Unstructured{}
+		err = json.Unmarshal(request.Object.Raw, o)
+		obj = o
+
+	case request.Name == "":
+		o := &unstructured.UnstructuredList{}
+		o.SetAPIVersion(request.Kind.Group + "/" + request.Kind.Version)
+		o.SetKind(request.Kind.Kind + "List")
+		err = c.List(ctx, o, client.InNamespace(request.Namespace))
+		obj = o
+		operation = "DELETECOLLECTION"
+
+	default:
+		o := &unstructured.Unstructured{}
+		o.SetAPIVersion(request.Kind.Group + "/" + request.Kind.Version)
+		o.SetKind(request.Kind.Kind)
+		err = c.Get(ctx, kutil.Key(request.Namespace, request.Name), o)
+		obj = o
+	}
+
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	entryLogger := gardenlogger.
+		NewFieldLogger(logger, "resource", fmt.Sprintf("%s/%s/%s", request.Kind.Group, request.Kind.Version, request.Kind.Kind)).
```

Comment:
Why are you suggesting this?

Reply:
Hm, I thought you used `request.Kind` here unintentionally, as you are always checking/working with `request.Resource` and also named the log key `"resource"`. I just thought, it would make more sense to also put the values of `request.Resource` into the log instead of `request.Kind`. But not strong opinion on this one... :)